src/fcntl/statx.rs

```diff
@@ -0,0 +1,350 @@
+use std::{mem, os::unix::prelude::RawFd};
+
+use libc::{self, c_int, c_uint, mode_t};
+
+use crate::{errno::Errno, NixPath, Result, sys::stat::{SFlag, Mode}, unistd::{Uid, Gid}, sys::time::TimeSpec};
+
+libc_bitflags!(
+    /// Configuration options for statx.
+    pub struct Flags: c_int {
+        /// Allow empty `pathname`, to receive status of the specified file descriptor
+        /// itself (not a directory entry) instead.
+        AT_EMPTY_PATH;
+
+        /// Do not automount the last component of the path being queried.
+        AT_NO_AUTOMOUNT;
+
+        /// Do not follow symlink in the last component of the path being queried.
+        AT_SYMLINK_NOFOLLOW;
+
+        /// Do whatever usual, non-`x` stat does. The default.
+        AT_STATX_SYNC_AS_STAT;
+
+        /// Force returned attributes to be up to date (synchonized).
+        /// May trigger data writeback.
+        /// May involve additional roudtrips for networked filesystems.
+        AT_STATX_FORCE_SYNC;
+
+        /// Don't synchronize, use cached data if possible. This may lead to the
+        /// returned data to be approximate.
+        AT_STATX_DONT_SYNC;
+    }
+);
+
+libc_bitflags!(
+    /// Configuration options for statx.
+    pub struct Mask: c_uint {
+        /// Want stx_mode & S_IFMT
+        STATX_TYPE;
+        /// Want stx_mode & ~S_IFMT
+        STATX_MODE;
+        /// Want stx_nlink
+        STATX_NLINK;
+        /// Want stx_uid
+        STATX_UID;
+        /// Want stx_gid
+        STATX_GID;
+        /// Want stx_atime
+        STATX_ATIME;
+        /// Want stx_mtime
+        STATX_MTIME;
+        /// Want stx_ctime
+        STATX_CTIME;
+        /// Want stx_ino
+        STATX_INO;
+        /// Want stx_size
+        STATX_SIZE;
+        /// Want stx_blocks
+        STATX_BLOCKS;
+        /// [All of the above]
+        STATX_BASIC_STATS;
+        /// Want stx_btime
+        STATX_BTIME;
+        /// Want stx_mnt_id (since Linux 5.8);
+        STATX_MNT_ID;
+        /// [All currently available fields]
+        STATX_ALL;
+    }
+);
+
+/// Attempt to retrieve stats of `pathname`. If `pathname` is relative, `dirfd`
+/// is used as starting directory for lookups. If `dirfs` is None, current
+/// directory is used.
+/// `pathname` may be empty string. But instead of specifying empty string
+/// literal, you are adviced to use zero-terminated `CStr` to avoid extra allocation 
```

Comment:
How about adding some examples to show usage?  That would be especially helpful for the "empty pathname" case.

Reply:
Good idea. Shall doc examples be marked as testable code (and assert something about typical external files e.g. `/dev/null`)?